tools/azsdk-cli/docs/cli-commands-guidelines.md

```diff
@@ -0,0 +1,416 @@
+6. [Examples](#examples)
+7. [Testing Guidelines](#testing-guidelines)
+
+## Command Hierarchy
+
+All CLI commands must follow a predefined top-level command hierarchy. Commands are organized into the following categories:
+
+### 1. **engsys** - Engineering Systems
+**Namespace:** `Azure.Sdk.Tools.Cli.Tools.EngSys`  
+**Command Group:** `SharedCommandGroups.EngSys`  
+**Verb:** `engsys`
+
+For CLI operations related to engineering systems tooling and services:
+- Code ownership validation and manipulation
+- Log analysis and processing
+- Test analysis and reporting
+- Cleanup operations
+- Internal engineering workflows
+
+**Examples:**
+- `engsys cleanup artifacts`
+- `engsys codeowners validate`
+- `engsys logs analyze`
+- `engsys test-analysis report`
+
+### 2. **package** - Package Operations
+**Namespace:** `Azure.Sdk.Tools.Cli.Tools.Package`  
+**Command Group:** `SharedCommandGroups.Package`  
+**Verb:** `package`  
+
+For operations at the SDK package level. The package group has further sub-grouping for better organization:
+
+#### Core Package Operations:
+- Build/compile SDK code
+- Generate source code
+- Release packages
+- Update version information
+- Validate packages
+
+#### Sub-Groups:
+
+##### **readme** - README Operations  
+For generating and updating README files:
+- Generate README files
+- Update README content
+- Validate README format
+- README documentation
+
+##### **sample** - Sample Operations
+For generating and updating SDK samples:
+- Generate new samples
+- Update existing samples
+- Validate sample code
+- Sample documentation
+
+##### **test** - Test Operations
+For creating, updating, and running tests for SDK packages:
+- Create new tests
+- Update existing tests
+- Run tests for specific packages
+- Test validation and reporting
+
+**Examples:**
+- `package build --package-path ./sdk/storage`
+- `package check-release-ready --package-name azure-core --version 1.0.0`
+- `package generate --typespec-project ./specification/storage`
+- `package readme generate --package-path ./sdk/compute`
+- `package readme update --package-path ./sdk/keyvault --section getting-started`
+- `package release --package-name azure-core`
+- `package sample generate --package-path ./sdk/compute`
+- `package sample update --package-path ./sdk/storage --sample-name basic-usage`
+- `package show-details`
+- `package test generate --package-path ./sdk/storage --test-type <type>`
+- `package test run --package-path ./sdk/keyvault --test-suite integration`
+- `package validate --package-path ./sdk/keyvault`
+
+### 3. **github** - GitHub Operations
+**Namespace:** `Azure.Sdk.Tools.Cli.Tools.GitHub`  
+**Command Group:** Custom (no predefined group)  
+**Verb:** `github`
+
+For operations related to GitHub:
+- Create pull requests
+- Check PR status
+- Manage labels
+- Repository operations
+
+**Examples:**
+- `github create-pr --title "Update SDK" --description "Description"`
+- `github get-pr-details --pr 123`
```

Comment:
I think these would be better under `github pr 123` and `github pr create`. Though is there any value add vs. just using the github CLI directly for these functions (at least for `get`)? I guess if we're exposing them as MCP tools we might as well bake them into the CLI...